nautobot/extras/querysets.py

```diff
@@ -123,7 +122,7 @@ def _get_config_context_filters(self):
         )
 
         if self.model._meta.model_name == "device":
-            base_query.add((Q(roles=OuterRef("device_role")) | Q(roles=None)), Q.AND)
+            base_query.add((Q(roles=OuterRef("role")) | Q(roles=None)), Q.AND)
```

Comment:
Should be able to pull this line out of the `if device... elif virtualmachine` blocks since it's now the same for both.